.github/workflows/docker-ci.yml

```diff
@@ -36,23 +39,35 @@ jobs:
           if_false: ${{ secrets.DOCKERHUB_USERNAME }}/${{ github.event.repository.name }}:latest
       -
         name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2
       -
         name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
       -
         name: Login to DockerHub
-        uses: docker/login-action@v1 
+        uses: docker/login-action@v2 
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
       -
-        name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
+        name: Build and push scratch image
+        id: docker_build_scratch
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64,linux/386,linux/arm/v7
+          push: true
+          tags: ${{ steps.imgtag.outputs.value }}
+          target: scratch
+          build-args: 'GIT_DESC=${{ steps.tagger.outputs.tag }}'
+      -
+        name: Build and push Alpine image
+        id: docker_build_alpine
+        uses: docker/build-push-action@v4
         with:
           context: .
           platforms: linux/amd64,linux/arm64,linux/386,linux/arm/v7
           push: true
           tags: ${{ steps.imgtag.outputs.value }}
```

Comment:
We need also to calcululate separate set of tags for apline, otherwise it will push two different images into the same tag. Right now it sets tag to `latest` for regular master build and `latest,v1.x.x` for release build. Lets make it `latest-alpine` or `latest-alpine,v1.x.x-alpine` for alpine too, so both images will be available in the registry.

Reply:
Nice suggestion! Added `tags: ${{ steps.imgtag.outputs.value }}-alpine` for the Alpine build.

Reply:
@tiaga On release this variable expands to `latest,v1.x.x` -- two tags separated by comma. So it won't work as expected: it will push alpine to `latest` and `v1.x.x-alpine` instead of `latest-alpine` and `v1.x.x-alpine`. You gonna need to adjust "Determine image tag type" task (`id: imgtag`) to add proper values for alpine or add another similar step with values ending with alpine suffix.

Reply:
As GitHub Actions are not really flexible, so I can only suggest using condition for tags on a building step. Or probably tags for Alpine image should be defined in a separate task.
If you have a better ideas, let's improve the code.

Reply:
Hm, I assume that just another `haya14busa/action-cond` step is easier in terms of code readability. Will update the code a bit later.

Reply:
Agree with that, thanks! Will be looking forward.

Reply:
Can be checked now.